forem/forem
# Use navigator.share on mobile devices

## What type of PR is this? (check all applicable)


## Description

We now have a tool for filtering UI based on the current runtime context so we can reliably separate Desktop vs Mobile UI to display some features we want on either one of them.

We had the [web-share-wrapper](https://www.npmjs.com/package/web-share-wrapper) documented for quite a while because we weren't able to have a bit more fine grain control over it, but this PR now makes use of it on mobile devices (where `navigator.share` API [is available]()).

## Related Tickets & Documents

Closes #9962

## QA Instructions, Screenshots, Recordings

1. Boot Forem locally
2. Make sure the actions on a Post continue to work the same as before in a Desktop Browser
3. Browse your local environment using a mobile device **if possible**
   - Instead of using anchor tag intents the menu should only display a single `Share` option
   - Tapping on this `Share` option should use the native share from the OS
   - Android and iOS should both work the same
   - Desktop browser DevTools "mobile simulator" won't work because the Forem instance will still detect the context as a "Desktop browser"

**Desktop view**

Share intents are always visible and _"Share Post via..."_ will be presented when `navigator.share` API is available

<img width="1235" alt="Screen Shot 2021-06-08 at 10 19 49" src="https://user-images.githubusercontent.com/6045239/121221741-17a53880-c843-11eb-9d96-fcebcb2e987b.png">

**Mobile view (native share within browser)**

![IMG_635951DCFF42-1](https://user-images.githubusercontent.com/6045239/121222761-0e689b80-c844-11eb-94d6-4d72821f6a7c.jpeg)

### UI accessibility concerns?

None

## Added tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

      or in a [forem.dev](http://forem.dev) post

## [optional] What gif best describes this PR or how it makes you feel?

![shrek donkey smile](https://media.giphy.com/media/Z9DoletURZoQAhldHw/giphy.gif)
